app/assets/javascripts/courseEdit/components/ComponentEdit.js

```diff
@@ -12,10 +14,29 @@ class ComponentEdit extends React.Component {
     this.setState({ component: nextProps.component })
   }
 
+  renderComponentImage(type) {
+    if (type == 'slide') {
```

Comment:
switch on the type of component - we're changing it to be integers so that we can use switch statements.